feat(carousel): update to v2 (#1725)
carbon-design-system/carbon-for-ibm-dotcom-website: src/pages/components/carousel.mdx

3. **Carousel counter:** Displays the current view number and the total view number in this carousel. The total number may change based on the breakpoint being viewed.
4. **Next icon button:** Advances the carousel to the next view, containing additional displayed content.

## Usage guidance

### When should I use a carousel?

- Use when you have multiple related content frames but screen real estate is limited.
- Use when you have exploratory content that isn’t necessary but helpful for the user.
- Use when you want to tease the user with additional related content.

### When should I not use a carousel?

- Do not use when content frames are not related.
- Do not use when users need to consume all carousel content – users are never required to advance the slides to see all content.

### Content

> - If you are using cards as your content, the recommended number of cards is a minimum of 5 cards and a maximum of 15 cards to avoid excessive clicking for the user.

Reviewer: ```suggestion
```

  Make sure all content in the carousel is related.

- Do not add content that is essential to a carousel. This includes navigational or wayfinding content links that would lead the user to a key resource.

- Prioritize the most important content at the beginning of the carousel, so that it is visible without needing to advance the carousel view.

- Carousel content should be simple and concise.

- Common content used in carousels includes use cases, news articles, client testimonials, and quotes.

> 

Reviewer: ```suggestion
- If you are using cards as your content, the recommended number of cards is a minimum of 5 cards and a maximum of 15 cards to avoid excessive clicking for the user.
```

Moving this down since it's specific to `cards`

## Behavior

Carousel does not have an auto-play or loop option -- the only way to move between pages is to interact directly with the pagination buttons.

> ### Accessibility note

Reviewer: ```suggestion
```

We can get rid of the `Accessibility note` header here and have it contained within `Behavior`

The reasoning for the lack of auto-advance options comes from accessibility standards,

> as per the [IBM accessibility site section 2.2.2](https://www.ibm.com/able/requirements/requirements/?version=7_2#2_2_2),

Reviewer: ```suggestion
as per the [IBM accessibility site section 2.2.2](https://www.ibm.com/able/requirements/requirements/?version=7_2#2_2_2):
```

> "for moving, blinking, scrolling, or auto-updating information, the user can pause, stop or hide it, or control the update frequency,"

Reviewer: ```suggestion
"for moving, blinking, scrolling, or auto-updating information, the user can pause, stop or hide it, or control the update frequency".
```

> which translates to only focusing on direct user interaction.

Reviewer: ```suggestion
To meet this standard, the carousel view is only updated by direct user interaction.
```
